File: 백준/Bronze/13300. 방 배정/방 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int[][] room = new int[2][7];
		int N = Integer.parseInt(st.nextToken());
		int K = Integer.parseInt(st.nextToken());
		for (int i = 0; i < N; i++) {
			st = new StringTokenizer(br.readLine());
			int s = Integer.parseInt(st.nextToken());
			int grade = Integer.parseInt(st.nextToken());
			room[s][grade]++;
		}
		int res = 0;
		for (int[] row : room) {
			for (int i = 1; i < 7; i++) {
				res += Math.ceil((double) row[i] / K);
			}
		}
		System.out.println(res);
	}
}

File: 백준/Bronze/13300. 방 배정/방 배정.py
def main():
  N, K = map(int,input().split())
  room = [[0] * 7 for _ in range(2)]
  for _ in range(N):
    S, Y = map(int,input().split())
    room[S][Y] += 1
  res = 0
  for i in room:
    for j in i:
      res += j // K
      if j % K:
        res += 1
  print(res)

if __name__ == "__main__":
  main()